executor/joiner.go

```diff
@@ -86,12 +86,11 @@ func newJoiner(ctx sessionctx.Context, joinType plan.JoinType,
 		ctx:          ctx,
 		conditions:   filter,
 		outerIsRight: outerIsRight,
-		maxChunkSize: ctx.GetSessionVars().MaxChunkSize,
 	}
 	colTypes := make([]*types.FieldType, 0, len(lhsColTypes)+len(rhsColTypes))
 	colTypes = append(colTypes, lhsColTypes...)
 	colTypes = append(colTypes, rhsColTypes...)
-	base.chk = chunk.NewChunkWithCapacity(colTypes, ctx.GetSessionVars().MaxChunkSize)
+	base.chk = chunk.NewFixedChunk(colTypes, ctx.GetSessionVars().InitChunkSize, ctx.GetSessionVars().MaxChunkSize)
```

Comment:
this `base.chk` maybe a question point...

if we set small initChunkSize, growslice will take into.

maybe need a better join result generate mechanism, like  #7433